Firmware/util.cpp

```diff
@@ -389,6 +389,29 @@ void gcode_level_check(uint16_t nGcodeLevel) {
     );
 }
 
+
+void active_sheet_check(uint8_t sheetIdx) {
+    if (oCheckModel == ClCheckModel::_None)
+        return;
+
+    int8_t activeSheetIdx = eeprom_read_byte(&(EEPROM_Sheets_base->active_sheet));
+    if (sheetIdx == activeSheetIdx)
+        return;
+
+    // SERIAL_ECHO_START;
+    // SERIAL_ECHOLNPGM("Printer G-code level differs from the G-code ...");
+    // SERIAL_ECHOPGM("actual  : ");
+    // SERIAL_ECHOLN(activeSheetIdx);
+    // SERIAL_ECHOPGM("expected: ");
+    // SERIAL_ECHOLN(sheetIdx);
+        
+    render_M862_warnings(
+        _T(MSG_GCODE_DIFF_SHEET_CONTINUE)
+        ,_T(MSG_GCODE_DIFF_SHEET_CANCELLED)
+        ,(uint8_t)oCheckMode
+    );
```

Comment:
I think these 2 choices may not be enough and / or may be confusing for the end user.

I've been toying with the idea of multiple choises here. A rough implementation as such: 
```

    if (oCheckMode == 1) { // Warning, stop print if user selects 'No'
        bool requestedSheetInitialized = eeprom_is_sheet_initialized(sheetIdx);
        uint8_t choice = lcd_show_multiscreen_message_with_choices_and_wait_P(_i("Current active sheet differs from the G-Code."), true, 0,
            _i("Cancel print."));
            _i("Ignore and continue with current sheet."),
            (requestedSheetInitialized ? _i("Requested sheet installed. Update EEPROM and continue.") : nullptr),
        if(choice == 2) {
            eeprom_update_byte(&(EEPROM_Sheets_base->active_sheet), sheetIdx); // and continue print
        }
        else if(choice == 0 || choice == 0xFF) { // User cancel or Timeout
            lcd_print_stop();
        }
    } else if (oCheckMode == 2) { // Strict, always stop print
        lcd_show_fullscreen_message_and_wait_P(_i("Current active sheet differs from the G-code. Print cancelled."));
        lcd_print_stop();
    }
```

This at least gives the user the added option for changing to the requested sheet on the spot instead of having to cancel and re-start the print.

The remaining problem is the lack of verbosity about the current sheet and the requested sheet. Being able to print the names would be great, but that would cost too much. Let allone the issue in doing so for other languages....

Oh, and screen realestate too!

Any thoughts on this?